### DISCUSSION TITLE
How can I setup a WrapDB system for the Trilobite libraries?

### DISCUSSION THREAD

**Discussion by dreamer-coding** [mesonbuild/meson#12517] [Q&A]
If it's possible, how can I set up a WrapDB system for the Trilobite Standard Component Libraries (TSCL)? All projects built using Meson and all hosted on GitHub.

**Comment by mitchgrout** [comment]
I'm assuming you're referring to the libraries like [trilo-xtype-c](https://github.com/trilobite-stdlib/trilo-xtype-c).

You could start by writing a `.wrap` file within your own project, and if that works out you could submit a pull request to https://github.com/mesonbuild/wrapdb to include it within the hosted website. Your wrap file might look like this...

```ini
# subprojects/trilo-xtype-c.wrap
[wrap-git]
url = https://github.com/trilobite-stdlib/trilo-xtype-c
directory = trilo-xtype-c-0.1.1
revision = 0.1.1
depth = 1

[provide]
trilo-xtype-c = trilo_xtype_c_dep 
```

AFAIK `meson wrap install` more-or-less looks up what version of a library is available within the wrapdb, and then downloads all the associated files into your `subprojects` directory.

**Reply by dreamer-coding** [reply]
So the task of adding a wrap is as easy as adding a wrap file in the wrapDB repo `subprojects` directory? assuming this is how pure meson-build libraries are added.
